src/Features/CSharp/Portable/RemoveUnnecessarySuppression/CSharpRemoveUnnecessarySuppressionDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.RemoveUnnecessarySuppression
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal class CSharpRemoveUnnecessarySuppressionDiagnosticAnalyzer : AbstractCodeStyleDiagnosticAnalyzer
```

Comment:
I think we can now merge `AbstractBuiltInCodeStyleDiagnosticAnalyzer` and `AbstractCodeStyleDiagnosticAnalyzer` into one type, but for consistency with existing analyzers:
```suggestion
    internal class CSharpRemoveUnnecessarySuppressionDiagnosticAnalyzer : AbstractBuiltInCodeStyleDiagnosticAnalyzer
```

Reply:
could we do that in another PR? :)

Reply:
It should be simple as file moves. I can try it as a last commit on the PR once you are done.

Reply:
Ah sorry, I thought this was for the previous comment. Sorry, this one is fine for a separate PR.

Reply:
Sure, go for it :)